terraform/services/zscaler-security-groups/main.tf

```diff
@@ -0,0 +1,53 @@
+### Get vpc reference
+module "vpc" {
+  source = "../../modules/vpc"
+  app    = var.app
+  env    = var.env
+}
+
+### public
+resource "aws_security_group" "zscaler_public" {
+  name        = "${var.app}-${var.env}-allow-zscaler-public"
+  description = "Allow public zscaler traffic"
+  vpc_id      = module.vpc.id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "zscaler_allow_public_0" {
+  security_group_id = aws_security_group.zscaler_public.id
+  cidr_ipv4         = "65.213.206.0/24"
+  from_port         = 0
+  ip_protocol       = -1
+  to_port           = 0
+}
+resource "aws_vpc_security_group_ingress_rule" "zscaler_allow_public_1" {
+  security_group_id = aws_security_group.zscaler_public.id
+  cidr_ipv4         = "208.250.57.0/24"
+  from_port         = 0
+  ip_protocol       = -1
+  to_port           = 0
+}
+### private
+resource "aws_security_group" "zscaler_private" {
+  name        = "${var.app}-${var.env}-allow-zscaler-private"
+  description = "Allow internet zscaler traffic private"
+  vpc_id      = module.vpc.id
+}
+
+variable "private_list" {
+  type        = list(string)
+  description = "List of names"
+  default = [
+    "10.223.126.64/27",
+    "10.223.126.96/28",
+    "10.220.126.64/27",
+    "10.220.126.96/28"
+  ]
+}
+resource "aws_vpc_security_group_ingress_rule" "zscaler_allow_private" {
+  for_each          = toset(var.prviate_list)
```

Comment:
I think this `for_each` is the way to go. Do this for the public group as well and add `private_cidrs` and `public_cidrs` to variables.tf without a default. We can then pass those variables in in the workflow with aws-params-env-action as at https://github.com/CMSgov/ab2d-bcda-dpc-platform/blob/f394bd501b6ac9f0f65248532602ee06960de7cf/.github/workflows/build-runner-images.yml#L33.

Reply:
This is looking good. Please set up the plan and apply workflows so we can check the output.